### DISCUSSION TITLE
Wall color correction

### DISCUSSION THREAD

**Discussion by protyposis** [awawa-dev/HyperHDR#769] [Q&A]
In case of a colored wall, the light and wall colors add up and lead to wrong colors. For example, white light on green wall makes the light look green, red on green looks yellow.

Philips Ambilight TVs have a color correction setting where you can set the wall color, and it rebalances the lights for better color reproduction. I understand this isn't perfectly possible due to the additive nature of colors, e.g., you can't make a green wall look white, but still it should be possible to optimize a little.

Has anybody tried that with HyperHDR? Do the color calibration settings already allow this - and if so, how?

**Comment by awawa-dev** [comment]
You have 2 LED color calibrations in the processing tab where you define can primaries. Some LED drivers also allow white color calibration, which can also be used to strengthen or weaken a given channel but only RGB. The simplest, naive solutions, for example: weakening the red and green channels when the wall is beige or yellow or blue if the background has a cool shade. Since it is probably most visible on white, you can try to make a mathematical conversion between standard D65 and a shade that resembles it on the wall, but it only looks simple in the math equations and whether it will actually help with what can be described as glare is difficult to predict, because it is a very complex topic and not only the color of the wall is important, but also the properties of the material.
